git/gitRemind_test.go

```diff
@@ -0,0 +1,258 @@
+package git
+
+import (
+	"canaveral/lib"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"reflect"
+	"testing"
+)
+
+func TestAddDeleteReminders(t *testing.T) {
```

Comment:
Usually go tests are named after the function (i think). For instance, you have a function DelReminder, so a unit test would be called TestDelReminder. (This would be the same for AddReminder, with its unit test be labeled TestAddReminder).

I think what you have here is leaning towards more of an integration test, which is definitely useful, so don't straight up delete it. To roll with what we've been doing thus far, I suggest either to reliable the test function to be TestGitReminders (to be more an integration test label) and add unit tests in as well, or simply refactor the tests to be unit tests (ie. split them up into multiple test functions).

Reply:
I know that unit tests typically only involve testing the functionality of a single function, and I usually do that, but I didn't think it made the most sense in this case. I think that to explain why it would be most helpful if I explain a bit more about how the remind functionality works, why it works that way, and then why I had to test these together. 

My original thought for storing the git reminders was that it would be easiest to have a map that maps files to reminders. However, the obvious problem is that there isn't a convenient/consistent way to ensure that the reminders map would persist between executions of canaveral. So making the map is trivial, but saving it and referencing it later is hard. So what I decided to do instead is save a file in the user's project directory called `.remind.json` which is a json file (hidden because the first character is `.`) that stores objects mapping file names to lists of reminders.

This works fairly well, although there are some quirks that can be improved in the future. The biggest one is that currently I haven't found a good way to only read in the reminders for a given file at a given time. Thankfully json files can be parsed directly into hashmaps so once I read in the file I can easily find the reminders for it upon commit, so it's fast, but this means that when I change a file's reminders I have to:

1. Read in the entire `.remind.json`
2. Append to the list of reminders for the specified file
3. Write back the entire `.remind.json` file upon completion of adding

I've made some optimizations that I can go into more if you'd like, but I think for the most part it's okay as I don't expect any project will have a huge reminder file, and if we get to a point where we have users and this is an issue for someone, we can find more complicated ways to refactor it then.

So, as for why this made it hard to test that add and delete features: testing deleting without using the AddReminders function essentially means completely reimplementing the add reminders function. In my unit tests, I keep all of the state uninitialized before and after execution so that subsequent calls to the same test can't impact future runs. However, this means that going into the TestDelReminder test, I would need to create and add to the json file. This basically means totally reimplementing add in the test, and then I'd have to find a way to test my test implementation of the add function, to ensure that if the Del function fails it's not actually a problem of reminders being stored incorrectly.

Hopefully this makes it a little more clear why I tested them together, it was really annoying to work through lol. That said, I can refactor them if you still think it's necessary.

Reply:
Yea that makes sense. If I read correctly, essentially in order to test removing reminders, you must first add the reminders, and vice versa. Thus, it makes the most sense to just test them together.

I'm okay with that